### PR TITLE
CLOUDP-81188: fix deletion of Atlas Cluster

### PR DESCRIPTION
**Pull request by crew-helper**
This fixes cluster deletion: uses the `spec.Name` instead of the resource `Name`.

Also moves the testing logic for cluster deletion to `afterRun` to avoid double calls to cluster removal. Then waits until the cluster is removed and removes the project

### CODE REVIEW

File: test/int/cluster_test.go [hunk 1]
```diff
@@ -59,21 +62,23 @@ var _ = Describe("AtlasCluster", func() {
...
+			Eventually(removeAtlasProject(createdProject.Status.ID), 600, interval).Should(BeTrue())
 		}
 
 		By("Removing the namespace " + namespace.Name)
 		err := k8sClient.Delete(context.Background(), &namespace)
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	Describe("Create/Update/Delete the cluster", func() {
+	FDescribe("Create/Update the cluster", func() {
```
Comment (vasilevp) [inline test/int/cluster_test.go:81]:
```suggestion
	Describe("Create/Update the cluster", func() {
```

File: pkg/controller/atlascluster/atlascluster_controller.go [hunk 2]
```diff
@@ -135,12 +135,12 @@ func (r *AtlasClusterReconciler) Delete(obj runtime.Object) error {
...
 	}
 
-	client, err := atlas.Client(connection, log)
+	atlasClient, err := atlas.Client(connection, log)
 	if err != nil {
 		return fmt.Errorf("cannot build Atlas client: %w", err)
 	}
 
-	_, err = client.Clusters.Delete(context.Background(), project.Status.ID, cluster.Name)
+	_, err = atlasClient.Clusters.Delete(context.Background(), project.Status.ID, cluster.Spec.Name)
```
Comment (vasilevp) [inline pkg/controller/atlascluster/atlascluster_controller.go:143]: :man_facepalming:

**Review by vasilevp — APPROVED** [review]

File: pkg/controller/atlascluster/atlascluster_controller.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func (r *AtlasClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	}
 	ctx := customresource.MarkReconciliationStarted(r.Client, cluster, log)
 
-	log.Infow("-> Starting AtlasCluster reconciliation", "spec", cluster.Spec)
+	log.Infow("-> Starting AtlasCluster reconciliation", "spec", cluster.Spec, "generation", cluster.Generation, "status", cluster.Status)
```
Comment (crew-helper) [inline pkg/controller/atlascluster/atlascluster_controller.go:61]: @vasilevp  this one was added for debugging issue - I expect there are sometimes caching problems and that's why our tests are flaky

File: pkg/util/testutil/customresources.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ func WaitFor(k8sClient client.Client, createdResource mdbv1.AtlasCustomResource,
 		}
 		// Atlas Operator hasn't started working yet
 		if createdResource.GetGeneration() != createdResource.GetStatus().GetObservedGeneration() {
+			fmt.Printf("Generation: %+v, observed Generation: %+v\n", createdResource.GetGeneration(), createdResource.GetStatus().GetObservedGeneration())
```
Comment (crew-helper) [inline pkg/util/testutil/customresources.go:25]: this is also for debugging of flaky tests

**Review by leo-ri — APPROVED** [review]

**Pull request merged**